server/src/main/java/org/opensearch/identity/noop/NoopSubject.java

```diff
@@ -48,4 +52,10 @@ public String toString() {
     public void login(final AuthenticationToken token) {
         // Noop subject is always logged in, and all authentication tokens are accepted
     }
+
+    @Override
+    public UnauthorizedException checkPermission(final List<String> permissions) {
+        System.err.println("Check for permission: " + permissions.stream().collect(Collectors.joining(", ")));
```

Comment:
why sys.err and not logger here?

Reply:
This is a debugging relic, I'll log in a cleaner way